src/transformers/modeling_utils.py

```diff
@@ -1530,7 +1530,9 @@ def _from_config(cls, config, **kwargs):
                 torch_dtype=torch_dtype,
             )
 
-        if is_deepspeed_zero3_enabled():
+        is_quantized = hasattr(config, "quantization_config")
+
+        if is_deepspeed_zero3_enabled() and not is_quantized:
```

Comment:
Not a huge fan of checking the `quantization_config` here as we don't really quantize the model with `from_config`. However, I'm not sure if there is an easier solution.Another solution would be to pass an arg in kwargs that we will pop.

Reply:
In case that there is  submodel that calls from_pretraned of the model that calls from_config. I just trying to ensure propagation of the quantization_config t all sub models that may need it.